bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -151,12 +147,9 @@ public void disconnected() {
         updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Disconnected");
 
         // retry connection if disconnect is not explicitly requested
-        if (conf != null && !explicitDisconnect && conf.connectionRetryInterval > 0) {
-            scheduler.schedule(() -> {
-                if (conn != null && !connecting && !conn.isConnected()) {
-                    connect();
-                }
-            }, conf.connectionRetryInterval, TimeUnit.SECONDS);
+        if (!explicitDisconnect && conf != null && conf.connectionRetryInterval > 0) {
+            logger.debug("Scheduling to reconnect in {} seconds.", conf.connectionRetryInterval);
+            scheduler.schedule(this::connect, conf.connectionRetryInterval, TimeUnit.SECONDS);
```

Comment:
A binding must not execute any task if it has been disposed. So, this task needs to be cancelled in `dispose()`.

Reply:
Added cancelling task on dispose